Hivemind/Assets/Scripts/GameWorld.cs

```diff
@@ -244,25 +261,27 @@ private IEnumerator LoadEnumerator(string name)
                 GameObject newNode = (GameObject)GameObject.Instantiate(Resources.Load($"Prefabs/Resources/{data.Prefab}"), new Vector3(data.PositionX, data.PositionY, data.PositionZ), Quaternion.identity);
                 newNode.GetComponent<ResourceNode>().SetData(data);
             }
-
-            foreach (UnitController controller in UnitControllerList)
-            {
-                controller.SetData(saveObject.TeamMindGroupData.FirstOrDefault(data => data.TeamId == controller.TeamId).MindGroupDataList);
-            }
-
             for (int i = 0; i < AntList.Count;)
             {
                 AntList[i].Destroy();
             }
+            foreach (UnitController controller in UnitControllerList)
+            {
+                controller.SetData(saveObject.TeamMindGroupData.FirstOrDefault(data => data.TeamId == controller.TeamId).MindGroupDataList);
+            }
             for (int i = 0; i < saveObject.Ants.Count; i++)
             {
                 AntData data = saveObject.AntData[i];
                 GameObject newAnt = (GameObject)GameObject.Instantiate(Resources.Load($"Prefabs/{data.Prefab}"), new Vector3(data.PositionX, data.PositionY, data.PositionZ), Quaternion.identity);
                 newAnt.GetComponent<Ant>().SetData(data);
             }
-            for (int i = 0; i < BaseControllerList.Count; i++)
+            foreach (BaseController controller in BaseControllerList)
+            {
+                controller.SetData(saveObject.BaseControllerData.FirstOrDefault(data => data.TeamID == controller.TeamID));
+            }
+            for (int i = 0; i < saveObject.BasicAi.Count; i++)
             {
-                BaseControllerList[i].SetData(saveObject.BaseControllerData[i]);
+                AIList[i].SetData(saveObject.BasicAIData[i]);
```

Comment:
Ik denk dat het beter is om te checken welke data bij welke AI hoort, het lijkt me niet verstandig om ervan uit te gaan dat ze zich iedere keer op dezelfde manier aanmelden